src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1107,6 +1097,18 @@ protected override void PreStart()
             _ackIdleTimerCancelable = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(ackIdleInterval, ackIdleInterval, Self, AckIdleCheckTimer.Instance, Self);
         }
 
+        private async Task<object> AssociateAsync()
+        {
+            try
+            {
+                return new Handle(await Transport.Associate(RemoteAddress, _refuseUid));
+            }
+            catch (Exception e)
+            {
+                return new Status.Failure(e.InnerException);
```

Comment:
This change looks good to me... I think `TaskScheduler.Default` will still get used here, but even if it doesn't I don't think it will matter given how simple the continuation is (object / exception mapping.)